addon/components/frost-button.js

```diff
@@ -241,9 +229,9 @@ export default Component.extend(SpreadMixin, PropTypeMixin, {
       return true
     }
 
-    const onClickHandler = this._getOnClickHandler()
-    if (onClickHandler && !get(this, 'disabled')) {
-      onClickHandler(get(this, 'id'))
+    const onClick = this.get('onClick')
+    if (typeOf(onClick) === 'function' && !this.get('disabled')) {
+      onClick(this.get('id'))
```

Comment:
I'm already fixing this in my PR, but I'm 99.9999% sure we decided that an explicit check for whether it's a function is unnecessary, and well as using `this.onClick` being fine instead of `this.get('onClick')`.